packages/typescript/src/index.ts

```diff
@@ -36,7 +36,7 @@ export default function typescript(options: RollupTypescriptOptions = {}): Plugi
   const watchProgramHelper = new WatchProgramHelper();
 
   const parsedOptions = parseTypescriptConfig(ts, tsconfig, compilerOptions, noForceEmit);
-  const filter = createFilter(include || ['*.ts+(|x)', '**/*.ts+(|x)'], exclude, {
+  const filter = createFilter(include || '{,**/}*.?(c|m)ts?(x)', exclude, {
```

Comment:
Could you add a comment listing the extensions this glob corresponds to (.ts, .tsx, .mts, .cts, etc) so its clear to future contributors?

Reply:
Yeah!

Reply:
Actually it also would include ctsx and mtsx, right? Is that worth mentioning?

Reply:
The thing is that those aren't officially supported extensions... I guess you could refine it so that they aren't included, but not sure that's worth the trouble...

Reply:
No, I would not suggest to change the pattern. I'm just wondering if the comment is more confusing than helpful if it does not list all matching extensions.

Reply:
Maybe add them in parenthesis?

@NotWoods, do you have an opinion on this?

Reply:
Could we redefine the glob? I think `'**/*.(cts|mts|ts|tsx)'` would work and be pretty clear.

Reply:
I think you'd still need the first curly group, but apart from that I second that suggestion. In this case, the explicit listing makes the pattern more readable, not less.

Reply:
I like the old syntax using more glob features but I changed to the simpler explicit version!